<a href="https://colab.research.google.com/github/CakeNuthep/Create_Knowledge_Graph/blob/main/Analyse_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

In [10]:
df_acthor_tagged = pd.read_csv('/content/drive/MyDrive/SuperAI/Knowledge Graph/Dataset/author-tagged.csv')
df_data_combine = pd.read_excel('/content/drive/MyDrive/SuperAI/Knowledge Graph/Dataset/DataCombine_version2.xlsx','วิเคราะห์')

In [11]:
df_acthor_tagged

Tname                      Ename
0             สุรีย์พร กฤษเจริญ      Sureeporn Kritcharoen
1              อรอนงค์ ดอกจันรี                           
2             วีรศักดิ์ ฟองเงิน          Weerasak Fongngen
3             สุรพงษ์ เพ็ชร์หาญ           Surapong Petharn
4                 ชัยวุฒิ โกเมศ              Chaivut Komez
...                         ...                        ...
203824    สุจิตรา ชัยกิตติศิลป์     Suchitra Chaikittisilp
203825           ดวงใจ ลิมตโสภณ        Duangjai Limtasopon
203826          อังสนา ศิรประชา        Aungsana Sirapracha
203827       บุบผา วิริยรัตนกุล     Buppar Viriyaratanakul
203828  แสงทอง เลิศประเสริฐพงศ์  Sangthong Lertprasertpong

[203829 rows x 2 columns]

In [12]:
df_data_combine

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
0            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/hafis...
1            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/asst-...
2            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/hasun...
3            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/hassa...
4            NaN  ...                                                NaN
...          ...  ...                                                ...
7299         NaN  ...  https://www.krai.io/people,https://set.ait.ac....
7300         NaN  ...  https://repository.nida.ac.th/handle/662723737...
7301         NaN  ...  https://pdfs.semanticscholar.org/be4e/ef39fc48...
7302         NaN  ...  http://www.eng.au.edu/index.php/about/faculty-...
7303         NaN  ...                  https://me.eng.cmu.ac.th/staff/47

[7304 rows x 24 columns]

In [81]:
last_name_index = 8
last_name = df_data_combine.columns[last_name_index]
first_name_index = 7
first_name = df_data_combine.columns[first_name_index]

In [83]:
df_data_combine[first_name] = df_data_combine[first_name].str.upper() 
df_data_combine[last_name] = df_data_combine[last_name].str.upper() 

# วิเคราะห์เรื่องการ Duplicate ข้อมูล

In [85]:
def get_count(values):
    return len(values)

In [86]:

grouped_count = df_data_combine.groupby(last_name).agg(get_count)

In [87]:
grouped_count

Unnamed: 0  ...  อ้างอิง (เว็บไซต์)
Last name                          ...                    
-                          1072.0  ...                1072
A. BAY                        1.0  ...                   1
A. HIPSHER                    1.0  ...                   1
ABDEL RAHMAN TAWALBEH         1.0  ...                   1
ABDULLAH                      1.0  ...                   1
...                           ...  ...                 ...
YUTTHASARNSENEE               1.0  ...                   1
YUVAPOOSITANON                1.0  ...                   1
YUWAKOSOL                     1.0  ...                   1
ZHAO                          1.0  ...                   1
๋JANTIMA                      1.0  ...                   1

[5749 rows x 23 columns]

In [88]:

grouped_count = df_data_combine.groupby(first_name).agg(get_count)

In [89]:
grouped_count

Unnamed: 0  Unnamed: 1  ...  ความเชี่ยวชาญ  อ้างอิง (เว็บไซต์)
First name                           ...                                   
-                1033.0        1033  ...           1033                1033
ABDULFATTAH         1.0           1  ...              1                   1
ABHISITH            1.0           1  ...              1                   1
ABIDENG             1.0           1  ...              1                   1
ACHANAI             1.0           1  ...              1                   1
...                 ...         ...  ...            ...                 ...
‪THINNAPHAT         1.0           1  ...              1                   1
‪VITTHAYA           1.0           1  ...              1                   1
‪WANCHAI            1.0           1  ...              1                   1
‪WANVIMOL           1.0           1  ...              1                   1
‪WIPHUSANA          1.0           1  ...              1                   1

[4426 rows x 23 columns]

In [90]:
grouped_count = df_data_combine.groupby([first_name,last_name]).agg(get_count)

In [91]:
grouped_count

Unnamed: 0  ...  อ้างอิง (เว็บไซต์)
First name  Last name                    ...                    
-           -                    1033.0  ...                1033
ABDULFATTAH MASAMEA                 1.0  ...                   1
ABHISITH    DOEMSUNTIA              1.0  ...                   1
ABIDENG     HAWA                    1.0  ...                   1
ACHANAI     BUASRI                  1.0  ...                   1
...                                 ...  ...                 ...
‪THINNAPHAT BORIRAK                 1.0  ...                   1
‪VITTHAYA   PHANCHAROENSINN         1.0  ...                   1
‪WANCHAI    CHANKAIPOL              1.0  ...                   1
‪WANVIMOL   NADEE                   1.0  ...                   1
‪WIPHUSANA  CHAYINTHU               1.0  ...                   1

[6051 rows x 22 columns]

In [92]:
duplicate_name = df_data_combine.duplicated([first_name,last_name])

In [94]:
df_duplicate_name = df_data_combine[duplicate_name]

In [95]:
df_duplicate_name

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
36           NaN  ...           https://ict.up.ac.th/personnel/major/ice
42           NaN  ...  https://baac.reru.ac.th/index.php?option=com_c...
46           NaN  ...              https://finearts.dpu.ac.th/staff/3/20
47           NaN  ...  http://www.mct.rmutt.ac.th/aboutus/personnel-n...
51           NaN  ...  http://idt.pnru.ac.th/index.php?op=detail&opid...
...          ...  ...                                                ...
7185         1.0  ...                                                NaN
7186         NaN  ...            https://ict.up.ac.th/personnel/major/cs
7189         NaN  ...  http://www.eng2.su.ac.th/department_industrial...
7208         NaN  ...     http://bba.ubru.ac.th/web/person.php?tid=11#r2
7270         NaN  ...  https://scholar.google.com/citations?user=xsYO...

[1253 rows x 24 columns]

In [96]:
df_duplicate_name.to_csv('duplicate_name.csv')

In [97]:
duplicate_and_duplicate_name = df_duplicate_name.duplicated([first_name,last_name])

In [98]:
df_duplicate_and_duplicate_name = df_duplicate_name[duplicate_and_duplicate_name == False]

In [99]:
df_duplicate_and_duplicate_name

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
36           NaN  ...           https://ict.up.ac.th/personnel/major/ice
66           1.0  ...   http://itech.tru.ac.th/2014/?page=teacher&id=125
104          1.0  ...                                                NaN
108          1.0  ...  https://engineer.utcc.ac.th/?p=personel&uid=am...
151          1.0  ...  https://www.cpe.eng.cmu.ac.th/academicView.php...
...          ...  ...                                                ...
7135         NaN  ...  https://cvs.enit.kku.ac.th/cv/profile/123551/%...
7171         1.0  ...  https://www.spu.ac.th/fac/engineer/th/staff.ph...
7180         1.0  ...                                                NaN
7185         1.0  ...                                                NaN
7270         NaN  ...  https://scholar.google.com/citations?user=xsYO...

[221 rows x 24 columns]

In [101]:
index_prefix_education = 2
index_prefix = 3
prefix_education = df_data_combine.columns[index_prefix_education]
prefix = df_data_combine.columns[index_prefix]

In [102]:
for index,row in df_duplicate_and_duplicate_name.iterrows():
  print(df_data_combine[(df_data_combine[first_name] == row[first_name]) & (df_data_combine[last_name] == row[last_name])][[prefix_education,prefix,first_name,last_name]])

     คำนำหน้า\nทางวิชาการ คำนำหน้า\nสถานภาพ First name Last name
25                อาจารย์               นาย          -         -
36     ผู้ช่วยศาสตราจารย์               นาย          -         -
42                อาจารย์            นางสาว          -         -
46                อาจารย์               นาย          -         -
47                อาจารย์               นาย          -         -
...                   ...               ...        ...       ...
7156              อาจารย์            นางสาว          -         -
7159          ศาสตราจารย์               นาย          -         -
7186              อาจารย์               นาย          -         -
7189              อาจารย์               นาย          -         -
7208              อาจารย์               นาย          -         -

[1033 rows x 4 columns]
   คำนำหน้า\nทางวิชาการ คำนำหน้า\nสถานภาพ First name Last name
65              อาจารย์               นาย        ITT     RANOK
66              อาจารย์               นาย        ITT     RANOK
    คำ

In [103]:
df_data_combine[(df_data_combine[first_name]=='PUTTHIPONG')]

Unnamed: 0  ...                            อ้างอิง (เว็บไซต์)
3125         NaN  ...  http://www.eng.src.ku.ac.th/user/putthipong/
3126         1.0  ...  http://www.eng.src.ku.ac.th/user/putthipong/

[2 rows x 24 columns]

In [106]:
df_copy = pd.DataFrame(columns=df_data_combine.columns)

In [107]:
df_copy

Empty DataFrame
Columns: [Unnamed: 0, Unnamed: 1, คำนำหน้า
ทางวิชาการ, คำนำหน้า
สถานภาพ, คำนำหน้า
ตำแหน่งงาน, ชื่อ, นามสกุล, First name, Last name, เพศ, ชื่อเต็มมหาวิทยาลัย, คณะ, สาขา, จัหวัดที่ตั้งมหาวิทยาลัย, อีเมล, เบอร์โทร, ชื่อสถาบันศึกษาระดับปริญญาตรี, วุฒิการศึกษาระดับปริญญาตรี, ชื่อสถาบันศึกษาระดับปริญญาโท, วุฒิการศึกษาระดับปริญญาโท, ชื่อสถาบันศึกษาระดับปริญญาเอก, วุฒิการศึกษาระดับปริญญาเอก, ความเชี่ยวชาญ, อ้างอิง (เว็บไซต์)]
Index: []

In [110]:
for index,row in df_duplicate_and_duplicate_name.iterrows():
  df_copy = df_copy.append(df_data_combine[(df_data_combine[first_name] == row[first_name]) & (df_data_combine[last_name] == row[last_name])])
  # print(df_data_combine[(df_data_combine[first_name] == row[first_name]) & (df_data_combine[last_name] == row[last_name])][[prefix_education,prefix,first_name,last_name]])

In [111]:
df_copy

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
25           1.0  ...                                                NaN
36           NaN  ...           https://ict.up.ac.th/personnel/major/ice
42           NaN  ...  https://baac.reru.ac.th/index.php?option=com_c...
46           NaN  ...              https://finearts.dpu.ac.th/staff/3/20
47           NaN  ...  http://www.mct.rmutt.ac.th/aboutus/personnel-n...
...          ...  ...                                                ...
7180         1.0  ...                                                NaN
7184         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
7185         1.0  ...                                                NaN
7267         NaN  ...  https://scholar.google.com/citations?user=xsYO...
7270         NaN  ...  https://scholar.google.com/citations?user=xsYO...

[1474 rows x 24 columns]

In [135]:
df_copy[df_copy[first_name] != '-']

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
65           NaN  ...   http://itech.tru.ac.th/2014/?page=teacher&id=125
66           1.0  ...   http://itech.tru.ac.th/2014/?page=teacher&id=125
103          NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
104          1.0  ...                                                NaN
107          NaN  ...  https://engineer.utcc.ac.th/?p=personel&uid=am...
...          ...  ...                                                ...
7180         1.0  ...                                                NaN
7184         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
7185         1.0  ...                                                NaN
7267         NaN  ...  https://scholar.google.com/citations?user=xsYO...
7270         NaN  ...  https://scholar.google.com/citations?user=xsYO...

[441 rows x 24 columns]

In [138]:
df_difference_prefix_education = pd.DataFrame(columns=df_data_combine.columns)
for index,row in df_duplicate_and_duplicate_name.iterrows():
  df_uniq = df_copy[(df_copy[first_name] == row[first_name]) & (df_copy[last_name] == row[last_name])]
  if len(df_uniq.groupby([prefix_education,first_name,last_name]))>1:
    df_difference_prefix_education = df_difference_prefix_education.append(df_uniq)


In [139]:
df_difference_prefix_education[df_difference_prefix_education[first_name] != '-']

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
150          NaN  ...  https://www.researchgate.net/profile/Arnan_Sip...
151          1.0  ...  https://www.cpe.eng.cmu.ac.th/academicView.php...
327          NaN  ...                                                NaN
328          NaN  ...                                                NaN
470          NaN  ...  https://sites.google.com/udru.ac.th/electrical...
471          1.0  ...  https://sites.google.com/udru.ac.th/electrical...
1108         1.0  ...                 http://www.it.kmitl.ac.th/~sirion/
1109         NaN  ...                                                NaN
1112         1.0  ...   http://itech.tru.ac.th/2014/?page=teacher&id=127
1113         NaN  ...   http://itech.tru.ac.th/2014/?page=teacher&id=127
1561         1.0  ...   http://itech.tru.ac.th/2014/?page=teacher&id=146
1562         NaN  ...   http://itech.tru.ac.th/2014/?page=teacher&id=146
1738         NaN  ...  http://en.cpru.ac.th/th/%e0%b8%9a%e0%b8%b8%e0%...
1739         NaN  ...  https://madlab.cpe.ku.ac.th/ThailandResearch/?...
1788         1.0  ...     http://www1.it.kmitl.ac.th/about/personnel/450
1789         NaN  ...                                                NaN
1857         1.0  ...  http://www.it.mut.ac.th/index.php/component/co...
1858         NaN  ...  http://www.it.mut.ac.th/index.php/component/co...
1958         1.0  ...   http://itech.tru.ac.th/2014/?page=teacher&id=130
1959         NaN  ...   http://itech.tru.ac.th/2014/?page=teacher&id=130
2051         1.0  ...   http://itech.tru.ac.th/2014/?page=teacher&id=151
2052         NaN  ...   http://itech.tru.ac.th/2014/?page=teacher&id=151
2096         NaN  ...  https://erp.mju.ac.th/person_detail.aspx?pid=M...
2097         1.0  ...  https://erp.mju.ac.th/person_detail.aspx?pid=M...
2529         1.0  ...         https://www.it.kmitl.ac.th/th/staff/lapas/
2530         NaN  ...                                                NaN
3096         1.0  ...  https://engineer.utcc.ac.th/?p=personel&uid=pa...
3097         NaN  ...  https://engineer.utcc.ac.th/?p=personel&uid=pa...
3536         NaN  ...  http://eve.eng.ku.ac.th/index.php/th/tea/175-p...
3537         1.0  ...  http://eve.eng.ku.ac.th/index.php/th/tea/175-p...
3559         1.0  ...        http://www.eng.src.ku.ac.th/user/pongsakon/
3560         NaN  ...        http://www.eng.src.ku.ac.th/user/pongsakon/
3674         1.0  ...     http://www1.it.kmitl.ac.th/about/personnel/462
3675         NaN  ...                                                NaN
3871         1.0  ...  http://www.me.eng.ku.ac.th/index.php?option=co...
3872         NaN  ...  http://www.me.eng.ku.ac.th/index.php?option=co...
3932         NaN  ...  https://research.ku.ac.th/ku_resume/?idcode=51...
3933         1.0  ...          http://www.eng.src.ku.ac.th/user/poranat/
4080         1.0  ...        https://www.it.kmitl.ac.th/th/staff/bundit/
4081         NaN  ...                                                NaN
4424         NaN  ...  http://www.next.cebuu.net/index.php/site-layou...
4425         1.0  ...  http://www.next.cebuu.net/index.php/site-layou...
4572         1.0  ...   http://itech.tru.ac.th/2014/?page=teacher&id=139
4573         NaN  ...   http://itech.tru.ac.th/2014/?page=teacher&id=139
4715         1.0  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
4716         NaN  ...                                                NaN
4877         1.0  ...  http://stdb.most.go.th/scientist_detail.aspx?i...
4878         NaN  ...  https://www.research.kmutnb.ac.th/pub/research...
5042         1.0  ...         http://www.ietech.kmitl.ac.th/?page_id=608
5043         NaN  ...  http://www.ietech.kmitl.ac.th/wp-content/uploa...
6096         NaN  ...  https://www.coe.phuket.psu.ac.th/faculty-of-te...
6097         1.0  ...                                                NaN
6221         1.0  ...     http://www1.it.kmitl.ac.th/about/personnel/730
6222         NaN  ...                                           

In [134]:
len(df_difference_prefix_education.groupby([first_name,last_name]))

31

In [136]:
df_difference_prefix = pd.DataFrame(columns=df_data_combine.columns)
for index,row in df_duplicate_and_duplicate_name.iterrows():
  df_uniq = df_copy[(df_copy[first_name] == row[first_name]) & (df_copy[last_name] == row[last_name])]
  if len(df_uniq.groupby([prefix,first_name,last_name]))>1:
    df_difference_prefix = df_difference_prefix.append(df_uniq)

In [137]:
df_difference_prefix[df_difference_prefix[first_name] !='-']

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
373          NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
374          1.0  ...                                                NaN
419          NaN  ...                                                  -
420          1.0  ...                                                  -
1338         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
1339         1.0  ...                                                NaN
1683         NaN  ...  https://research.ku.ac.th/ku_resume/?idcode=48...
1684         1.0  ...  https://research.ku.ac.th/forest/Person.aspx?i...
2258         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
2259         1.0  ...                                                NaN
3125         NaN  ...       http://www.eng.src.ku.ac.th/user/putthipong/
3126         1.0  ...       http://www.eng.src.ku.ac.th/user/putthipong/
3815         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
3816         1.0  ...                                                NaN
3996         NaN  ...  http://www.ie.engr.tu.ac.th/lecturer/profile/b...
3997         1.0  ...  http://www.ie.engr.tu.ac.th/lecturer/profile/b...
4979         NaN  ...  http://tech.siamtechu.net/profile_teacher.php?...
4980         NaN  ...  https://sci.chandra.ac.th/newsci/index.php/off...
5417         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
5418         1.0  ...                                                NaN
5606         1.0  ...  http://www.sct.nstru.ac.th/sct/uploadFile/prof...
5607         NaN  ...  https://race.nstru.ac.th/home_ex/e-portfolio/i...
5662         1.0  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
5663         NaN  ...                                                NaN
5823         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
5824         1.0  ...                                                NaN
6134         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
6135         1.0  ...                                                NaN
7179         NaN  ...  http://www.ce.kmitl.ac.th/faculty.php?action=v...
7180         1.0  ...                                                NaN

[30 rows x 24 columns]

In [133]:
len(df_difference_prefix.groupby([first_name,last_name]))

16

## สรุปผล เรื่องการ Duplicate (จะเน้นไปเรื่องชื่อ นามสกุล ภาษาอังกฤษที่ Duplicate)
  - มีจำนวนคนที่ Uniq ตามชื่อและนามสกุลที่เป็นภาษาอังกฤษ 6051 คน
  - คนที่มีการ Duplicate ทั้งหมดมีจำนวน 221 คนจาก 6051 คน
	- ปัญหาคำนำหน้าทางวิชาการไม่เหมือนกัน (มี 31 คนจาก 221 คน)
	- ปัญหานาย นางกับนางสาว และ - (มี 16 คนจาก 221 คน) จากการสังเกตุเพิ่มเติมคาดว่าการใส่คอลัมที่ชื่อว่า เพศ เกิดจาก การดูคอมลัมน์คำนำหน้า

# วิเคราะห์เรื่อง ชื่อสถาบันศึกษาระดับปริญญาตรี  โท เอก บางครั้งเป็นชื่อภาษาไทยบางครั้งเป็นชื่อภาษาอังกฤษ

In [140]:
df_data_combine

Unnamed: 0  ...                                 อ้างอิง (เว็บไซต์)
0            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/hafis...
1            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/asst-...
2            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/hasun...
3            NaN  ...  http://eng.pnu.ac.th/web/academic-staffs/hassa...
4            NaN  ...                                                NaN
...          ...  ...                                                ...
7299         NaN  ...  https://www.krai.io/people,https://set.ait.ac....
7300         NaN  ...  https://repository.nida.ac.th/handle/662723737...
7301         NaN  ...  https://pdfs.semanticscholar.org/be4e/ef39fc48...
7302         NaN  ...  http://www.eng.au.edu/index.php/about/faculty-...
7303         NaN  ...                  https://me.eng.cmu.ac.th/staff/47

[7304 rows x 24 columns]

In [155]:
education_bachelor_index = 16
education_bachelor = df_data_combine.columns[education_bachelor_index]
education_master_index = 18
education_master = df_data_combine.columns[education_master_index]
education_doctor_index = 20
education_doctor = df_data_combine.columns[education_doctor_index]

In [156]:
import re

def IsEnglish(word):
    reg = re.compile(r'[a-zA-Z0-9 .!?"-]+')
    if pd.notnull(word):
      if reg.match(word):
        return True
      else:
        return False
    else:
      return True


In [157]:
df_data_combine[educcation_bachelor]

0       มหาวิทยาลัยนราธิวาสราชนครินทร์
1             มหาวิทยาลัยสงขลานครินทร์
2          มหาวิทยาลัยเทคโนโลยีสุรนารี
3                   มหาวิทยาลัยศรีปทุม
4                     วิทยาลัยครูสงขลา
                     ...              
7299           มหาวิทยาลัยยอนเซกรุงโซล
7300                                 -
7301                               NaN
7302             Assumption University
7303                                 -
Name: ชื่อสถาบันศึกษาระดับปริญญาตรี, Length: 7304, dtype: object

In [164]:
education_bachelor_column_name = 'education_bachelor_english'
df_data_combine[education_bachelor_column_name] = df_data_combine[education_bachelor].apply(IsEnglish)

education_master_column_name = 'education_master_english'
df_data_combine[education_master_column_name] = df_data_combine[education_master].apply(IsEnglish)

education_doctor_column_name = 'education_doctor_english'
df_data_combine[education_doctor_column_name] = df_data_combine[education_doctor].apply(IsEnglish)

## ปริญญาตรี

In [165]:
df_data_combine[[education_bachelor,education_bachelor_column_name]]

ชื่อสถาบันศึกษาระดับปริญญาตรี  education_bachelor_english
0     มหาวิทยาลัยนราธิวาสราชนครินทร์                       False
1           มหาวิทยาลัยสงขลานครินทร์                       False
2        มหาวิทยาลัยเทคโนโลยีสุรนารี                       False
3                 มหาวิทยาลัยศรีปทุม                       False
4                   วิทยาลัยครูสงขลา                       False
...                              ...                         ...
7299         มหาวิทยาลัยยอนเซกรุงโซล                       False
7300                               -                        True
7301                             NaN                        True
7302           Assumption University                        True
7303                               -                        True

[7304 rows x 2 columns]

In [177]:
#ไม่มีข้อมูลปริญญาตรี
df_data_combine[(df_data_combine[education_bachelor]=='-') | 
                (df_data_combine[education_bachelor].isna())]

Unnamed: 0 Unnamed: 1  ... education_master_english education_doctor_english
11           NaN         AI  ...                     True                     True
12           NaN         IT  ...                     True                     True
21           NaN         AI  ...                     True                     True
28           NaN         IT  ...                     True                     True
30           NaN         IT  ...                    False                     True
...          ...        ...  ...                      ...                      ...
7289         NaN         IT  ...                     True                     True
7296         NaN         IT  ...                     True                     True
7300         NaN         AI  ...                     True                    False
7301         NaN         AI  ...                     True                     True
7303         NaN         AI  ...                     True                     True

[1917 rows x 28 columns]

In [181]:
#มีข้อมูลปริญญาตรี
df_data_combine[~((df_data_combine[education_bachelor]=='-') | 
                (df_data_combine[education_bachelor].isna()))]

Unnamed: 0 Unnamed: 1  ... education_master_english education_doctor_english
0            NaN         IT  ...                    False                     True
1            NaN         AI  ...                     True                     True
2            NaN         IT  ...                    False                     True
3            NaN         AI  ...                    False                     True
4            NaN         IT  ...                    False                     True
...          ...        ...  ...                      ...                      ...
7295         NaN         IT  ...                     True                     True
7297         NaN         IT  ...                     True                     True
7298         NaN         AI  ...                     True                     True
7299         NaN         AI  ...                     True                     True
7302         NaN         AI  ...                     True                     True

[5387 rows x 28 columns]

In [182]:
df_have_bachelor = df_data_combine[~((df_data_combine[education_bachelor]=='-') | 
                (df_data_combine[education_bachelor].isna()))]

In [185]:
#ตรวจสอบว่ามีภาษาอังกฤษเท่าไหร่ ภาษาไทยเท่าไหร่
df_have_bachelor.groupby(education_bachelor_column_name).size()

education_bachelor_english
False    4483
True      904
dtype: int64

## ปริญญาโท

In [183]:
df_have_bachelor[[education_master,education_master_column_name]]

ชื่อสถาบันศึกษาระดับปริญญาโท  education_master_english
0     มหาวิทยาลัยเทคโนโลยีพระจอมเกล้าพระนครเหนือ                     False
1      International Islamic University Malaysia                      True
2                    มหาวิทยาลัยเทคโนโลยีสุรนารี                     False
3              มหาวิทยาลัยอิสลามนานาชาติมาเลเซีย                     False
4                       มหาวิทยาลัยสงขลานครินทร์                     False
...                                          ...                       ...
7295          The State University of New Jersey                      True
7297       Massachusetts Institute of Technology                      True
7298         State University of Ghent, Belgium.                      True
7299                                MSTU, Moscow                      True
7302                       Assumption University                      True

[5387 rows x 2 columns]

In [186]:
#ไม่มีข้อมูลปริญญาโท
df_data_combine[(df_data_combine[education_master]=='-') | 
                (df_data_combine[education_master].isna())]

Unnamed: 0 Unnamed: 1  ... education_master_english education_doctor_english
11           NaN         AI  ...                     True                     True
12           NaN         IT  ...                     True                     True
21           NaN         AI  ...                     True                     True
28           NaN         IT  ...                     True                     True
34           NaN         IT  ...                     True                     True
...          ...        ...  ...                      ...                      ...
7280         NaN         IT  ...                     True                     True
7283         NaN         IT  ...                     True                     True
7300         NaN         AI  ...                     True                    False
7301         NaN         AI  ...                     True                     True
7303         NaN         AI  ...                     True                     True

[1963 rows x 28 columns]

In [187]:
#มีข้อมูลปริญญาโท
df_data_combine[~((df_data_combine[education_master]=='-') | 
                (df_data_combine[education_master].isna()))]

Unnamed: 0 Unnamed: 1  ... education_master_english education_doctor_english
0            NaN         IT  ...                    False                     True
1            NaN         AI  ...                     True                     True
2            NaN         IT  ...                    False                     True
3            NaN         AI  ...                    False                     True
4            NaN         IT  ...                    False                     True
...          ...        ...  ...                      ...                      ...
7296         NaN         IT  ...                     True                     True
7297         NaN         IT  ...                     True                     True
7298         NaN         AI  ...                     True                     True
7299         NaN         AI  ...                     True                     True
7302         NaN         AI  ...                     True                     True

[5341 rows x 28 columns]

In [188]:
df_have_master = df_data_combine[~((df_data_combine[education_master]=='-') | 
                (df_data_combine[education_master].isna()))]

In [189]:
#ตรวจสอบว่ามีภาษาอังกฤษเท่าไหร่ ภาษาไทยเท่าไหร่
df_have_master.groupby(education_master_column_name).size()

education_master_english
False    3685
True     1656
dtype: int64

## ปริญญาเอก

In [168]:
df_data_combine[[education_doctor,education_doctor_column_name]]

ชื่อสถาบันศึกษาระดับปริญญาเอก  education_doctor_english
0                                             -                      True
1                                             -                      True
2                                             -                      True
3     International Islamic University Malaysia                      True
4                                             -                      True
...                                         ...                       ...
7299          Moscow State Technical University                      True
7300               สถาบันบัณฑิตพัฒนบริหารศาสตร์                     False
7301                                        NaN                      True
7302  Dalhousie University, Halifax, NS, Canada                      True
7303                                          -                      True

[7304 rows x 2 columns]

In [190]:
#ไม่มีข้อมูลปริญญาเอก
df_data_combine[(df_data_combine[education_doctor]=='-') | 
                (df_data_combine[education_doctor].isna())]

Unnamed: 0 Unnamed: 1  ... education_master_english education_doctor_english
0            NaN         IT  ...                    False                     True
1            NaN         AI  ...                     True                     True
2            NaN         IT  ...                    False                     True
4            NaN         IT  ...                    False                     True
6            NaN         IT  ...                    False                     True
...          ...        ...  ...                      ...                      ...
7273         NaN         IT  ...                     True                     True
7275         NaN         IT  ...                     True                     True
7294         NaN         IT  ...                     True                     True
7301         NaN         AI  ...                     True                     True
7303         NaN         AI  ...                     True                     True

[3732 rows x 28 columns]

In [191]:
#มีข้อมูลปริญญาเอก
df_data_combine[~((df_data_combine[education_doctor]=='-') | 
                (df_data_combine[education_doctor].isna()))]

Unnamed: 0 Unnamed: 1  ... education_master_english education_doctor_english
3            NaN         AI  ...                    False                     True
5            NaN         IT  ...                     True                    False
7            NaN         IT  ...                    False                    False
10           NaN         IT  ...                    False                    False
11           NaN         AI  ...                     True                     True
...          ...        ...  ...                      ...                      ...
7297         NaN         IT  ...                     True                     True
7298         NaN         AI  ...                     True                     True
7299         NaN         AI  ...                     True                     True
7300         NaN         AI  ...                     True                    False
7302         NaN         AI  ...                     True                     True

[3572 rows x 28 columns]

In [192]:
df_have_doctor = df_data_combine[~((df_data_combine[education_doctor]=='-') | 
                (df_data_combine[education_doctor].isna()))]

In [193]:
#ตรวจสอบว่ามีภาษาอังกฤษเท่าไหร่ ภาษาไทยเท่าไหร่
df_have_doctor.groupby(education_doctor_column_name).size()

education_doctor_english
False    1357
True     2215
dtype: int64

## สรุปเรื่อง ชื่อสถาบันศึกษาระดับปริญญาตรี  โท เอก บางครั้งเป็นชื่อภาษาไทยบางครั้งเป็นชื่อภาษาอังกฤษ
- จาก Row ทั้งหมด 7304 row
- มีข้อมูลสถาบันศึกษาระดับปริญญาตรีทั้งหมด 5387
    - มีชื่อที่เป็นภาษาไทย 4483
    - มีชื่อที่เป็นภาษาอังกฤษ 904
- มีข้อมูลสถาบันศึกษาระดับปริญญาโททั้งหมด 5341
    - มีชื่อที่เป็นภาษาไทย 3685
    - มีชื่อที่เป็นภาษาอังกฤษ 1656
- มีข้อมูลสถาบันศึกษาระดับปริญญาเอกทั้งหมด 3572
    - มีชื่อที่เป็นภาษาไทย 1357
    - มีชื่อที่เป็นภาษาอังกฤษ 2215

# สรุปทั้งหมด
**สิ่งที่พบ**
- ชื่อ Duplicate กัน
- บางครั้งเป็น - บางครั้งเป็นค่า NaN ในการระบุว่าไม่มีข้อมูล
- ชื่อสถาบันศึกษาระดับปริญญาตรี  โท เอก บางครั้งเป็นชื่อภาษาไทยบางครั้งเป็นชื่อภาษาอังกฤษ

- วุฒิการศึกษาระดับปริญญาตรี โท เอก บางครั้งเป็นชื่อย่อ บางครั้งเป็นชื่อเต็ม และบางครั้งเป็นภาษาอังกฤษ และอาจมีคำอื่นเข้ามาเช่น "กำลังศึกษา" , "ลา", "ปริญญาเอก"

## สรุปผล เรื่องการ Duplicate (จะเน้นไปเรื่องชื่อ นามสกุล ภาษาอังกฤษที่ Duplicate)
  - มีจำนวนคนที่ Uniq ตามชื่อและนามสกุลที่เป็นภาษาอังกฤษ 6051 คน
  - คนที่มีการ Duplicate ทั้งหมดมีจำนวน 221 คนจาก 6051 คน
	- ปัญหาคำนำหน้าทางวิชาการไม่เหมือนกัน (มี 31 คนจาก 221 คน)
	- ปัญหานาย นางกับนางสาว และ - (มี 16 คนจาก 221 คน) จากการสังเกตุเพิ่มเติมคาดว่าการใส่คอลัมที่ชื่อว่า เพศ เกิดจาก การดูคอมลัมน์คำนำหน้า

## สรุปเรื่อง ชื่อสถาบันศึกษาระดับปริญญาตรี  โท เอก บางครั้งเป็นชื่อภาษาไทยบางครั้งเป็นชื่อภาษาอังกฤษ
- จาก Row ทั้งหมด 7304 row
- มีข้อมูลสถาบันศึกษาระดับปริญญาตรีทั้งหมด 5387
    - มีชื่อที่เป็นภาษาไทย 4483
    - มีชื่อที่เป็นภาษาอังกฤษ 904
- มีข้อมูลสถาบันศึกษาระดับปริญญาโททั้งหมด 5341
    - มีชื่อที่เป็นภาษาไทย 3685
    - มีชื่อที่เป็นภาษาอังกฤษ 1656
- มีข้อมูลสถาบันศึกษาระดับปริญญาเอกทั้งหมด 3572
    - มีชื่อที่เป็นภาษาไทย 1357
    - มีชื่อที่เป็นภาษาอังกฤษ 2215